src/routes/settings/settings.tsx

```diff
@@ -1,20 +1,24 @@
-import { Button } from "../../components"
-
-export const Settings: React.FC = () => {
-  return (
-    <div className="flex flex-col bg-lightGray items-center pb-11">
-        <Button
-          variant="text"
-          className="opacity-100 hover:opacity-60 text-base max-w-max pt-0"
-          buttonText="Log out"
-          onClick={() => alert("Log out")}
-        />
-        <Button
-          variant="text"
-          className="text-red opacity-100 hover:opacity-60 text-base max-w-max"
-          buttonText="Delete profile"
-          onClick={() => alert("Profile deleted!")}
-        />
-    </div>
-  )
-}
\ No newline at end of file
+import { Button } from "../../components";
+import { useAuth } from "../../hooks/useAuth";
+
+export const Settings: React.FC = () => {
+  const handleLogoutSuccess = () => console.log("Logged Out");
+  const auth = useAuth({ onLogoutSuccess: handleLogoutSuccess });
+
+  return (
+    <div className="flex flex-col bg-lightGray items-center pb-11">
+      <Button
+        variant="text"
+        className="opacity-100 hover:opacity-60 text-base max-w-max pt-0"
+        buttonText="Log out"
+        onClick={auth.logout}

```

Comment:
This may be out of scope for this task, but I noticed that when logging out, the dashboard will still show the users name until the page is refreshed. Probably need to trigger a re-render in some way after the logout is done, or reroute back to login/main if dashboard is going to be a protected route.

Reply:
Yea, I figured the logout will probably reroute to login eventually so re-rendering seemed unnecessary. I guess I could implement the the reroute myself and merge it into the logout action branch (or vice versa).